sqlglot/generator.py

```diff
@@ -1267,21 +1282,35 @@ def expressions(self, expression, key=None, flat=False, indent=True, sep=", "):
             return ""
 
         if flat:
-            return sep.join(self.sql(e) for e in expressions)
+            return sep.join(self.maybe_comment(self.sql(e), e.comment) for e in expressions)
 
-        sql = (self.sql(e) for e in expressions)
-        # the only time leading_comma changes the output is if pretty print is enabled
+        sql_with_comment = [(self.sql(e), e.comment) for e in expressions]
+        num_sqls = len(sql_with_comment)
+
+        # The only time leading_comma changes the output is if pretty print is enabled
         if self._leading_comma and self.pretty:
             pad = " " * self.pad
             expressions = "\n".join(
-                f"{sep}{s}" if i > 0 else f"{pad}{s}" for i, s in enumerate(sql)
+                f"{sep if i > 0 else pad}{self.maybe_comment(sql, com, single_line=True)}"
+                for i, (sql, com) in enumerate(sql_with_comment)
+            )
+        elif self.pretty:
+            sep = sep.strip()
```

Comment:
why do you call sep.strip() what's going on here?

Reply:
That would happen before too in pretty mode, due to `sep()` -- it's so we don't have a trailing space after each comma if we're about to drop a newline right after it.